Update dependencies to allow building on ocaml 4.08.1

Cjen1 opened the pull request:
Updates dependencies such that it builds on ocaml-4.08.
Adds ocamlformat support

Requires testing

Potential places of issue:
 - Randomness may not be correctly initialised
 - Logic may be missing due to ignoring unused variable warnings (indicated via `_` before name)

(Also let me know what target branch to use, rather than me-box:0.5.3-dev)

Cjen1 committed:
Update docker repo, and mirage-net-psock dependency

Cjen1 committed:
Update ocaml/opam-dev repo to ocaml/opam2. Add OPAMYES environmnet variable

Cjen1 committed:
dunise project

Cjen1 committed:
Add ocamlformat

Cjen1 committed:
Apply formatting changes

Cjen1 committed:
Update dune dependencies

Cjen1 committed:
Update module/function names:
 B64 -> Base64
 Re_posix -> Re.Posix
 V4.pp_hum -> V4.pp
 Macaddr.of_bytes -> Macaddr.of_octets
 Pervasives.compare -> Stdlib.compare
 V4.Prefix.pp_hum -> V4.Prefix.pp
 Ethif_packet -> Ethernet_packet
 Ethif_wire -> Ethernet_wire
 Re_str -> Re.Str
 Cohttp_lwt_body -> Cohttp_lwt.Body

Cjen1 committed:
Update policy.ml to use variable name shadowing and standard naming scheme

Cjen1 committed:
Update lib/nat.ml to remove compiler warnings

Cjen1 committed:
Update frame parsing to use result types rather than old option type.

Cjen1 committed:
lib/env.ml update deps, clean up logic

Cjen1 committed:
Update formatting lib/frame.ml

Cjen1 committed:
Update lib/intf.ml for api changes, specifically for Ethif.write

Cjen1 committed:
Update lib/junction.ml formatting

Cjen1 committed:
Format lib/policy.ml

Cjen1 committed:
Update lib/server.ml

Cjen1 committed:
Update dns, use the dns.1.1.3 package

Cjen1 committed:
Update lib/junction.ml

Cjen1 committed:
Update lib/junction.ml

Cjen1 committed:
Update lib/interfaces.ml

Cjen1 committed:
Set header_size on lib/intf.ml

Cjen1 committed:
Use Random.int for packet id. Init in bin/core_network

Cjen1 committed:
Add catch_alls to allow building

Cjen1 committed:
Update core-network.export to reflect local build environment

Cjen1 committed:
Update docker images to build

Cjen1 committed:
Fix spelling of Dispatcher message

Cjen1 committed:
Update mirage-net-psock to use me-box repo

mor1 commented on Dockerfile-aarch64:
```diff
@@ -3,7 +3,7 @@ FROM jptmoore/alpine-baseimage-aarch64
 RUN sudo apk add libffi-dev libressl-dev linux-headers bash autoconf
 RUN opam remove ezirmin irmin re.1.7.1 bos lwt
-RUN opam pin add -n mirage-net-psock.0.1.0 https://github.com/sevenEng/mirage-net-psock.git
+RUN opam pin add -n mirage-net-psock.0.1.0 https://github.com/sevenEng/mirage-net-psock.git#921900d502504ac46ef63b52935e4398d24647f4 -y
```
Sometimes you're pinning `mirage-net-psock` to `sevenEng`, sometimes to `me-box` -- is there a reason?

Cjen1 replied:
The most recent version appears to be the me-box, but yes should pin to just me-box so will grep to fix that.

mor1 commented on Dockerfile-relay:
```diff
@@ -1,11 +1,13 @@
-RUN sudo apk update && sudo apk add alpine-sdk bash gmp-dev perl autoconf linux-headers &&\
-    #opam remote add git https://github.com/ocaml/opam-repository.git &&\
-    opam pin add -n mirage-net-psock.0.1.0 https://github.com/sevenEng/mirage-net-psock.git &&\
+ENV OCAMLYES=true
```
`OPAMYES`...? (and probably elsewhere -- I might not catch them all:)

mor1 commented on Dockerfile-relay-aarch64:
```diff
@@ -2,11 +2,11 @@ FROM jptmoore/alpine-baseimage-aarch64
 
 ADD core-network.export core-network.export
-RUN opam switch import core-network.export
+RUN opam switch import core-network.export -y
```
`-y` not required with `OPAMYES`

mor1 reviewed (changes requested):
Modulo any review comments, generally looks ok to me, assuming the Travis failures can be resolved.

Cjen1 committed:
Fix final bug to allow tests to pass

Lwt.pick -> Lwt.choose,

Random.int has maximum upper limit of 2^30 (only 16 bits needed for ipv4
identity)

Cjen1 committed:
Force tester to rebuild container

Cjen1 committed:
Consistency on OPAMYES and mirage-net-psock repo

Cjen1 commented:
Still to do is to update aarch64 images
